source/libraries/ContractInterfacesGenerator.ts

```diff
@@ -0,0 +1,191 @@
+import * as fs from "async-file";
+import { Abi, AbiFunction, Primitive } from 'ethereum';
+import { ContractCompiler } from "./ContractCompiler";
+import { CompilerOutput, CompilerOutputContracts } from "solc";
+import { Configuration } from './Configuration';
+
+
+export class ContractInterfaceGenerator {
+    private readonly compiler: ContractCompiler;
+    private readonly configuration: Configuration;
+
+    public constructor(configuration: Configuration, compiler: ContractCompiler) {
+        this.compiler = compiler;
+        this.configuration = configuration;
+    }
+
+    public async generateContractInterfaces(): Promise<String> {
+
+        const contractsOutput: CompilerOutput = await this.compiler.compileContracts();
```

Comment:
As part of some upcoming work I'll be doing I am going to extract out the logic that filters down to just contracts that are interesting (meaning none of the libraries or interface contracts).  Once that is done we should utilize it here so as well so we don't generate these for various things that are never instantiated like interfaces and libraries.  Just a thought to keep in mind and to comment on if you think that is a bad idea.  Nothing needs to be done in this PR though.